source/docs/magento1/search-engine.md

```diff
@@ -82,6 +90,146 @@ query Search {
 If you are using the default theme or the theme Chocolatine, the search bar
 should now be visible.
 
+<blockquote class="warning">
+⚠️ Common mistake: For magento 2 the elasticsuite does not index cms pages. To do that you need to install <a href="https://github.com/Smile-SA/magento2-module-elasticsuite-cms-search" target="_blank">Smile elasticsuite cms search module</a> on your magento instance. Failing to do so will cause the top search bar to malfunction. If you do not need the CMS Page search you can always override related `SearchBar` Components/GraphQL queries to disable the cms page search functionality as in the following diff:
+
+<details>
+<summary>
+Disable CMS Pages Search Diff
+</summary>
+
+#### Base theme:
+
+```
+diff --git a/src/web/theme/modules/Search/SearchBar/Autocomplete/Autocomplete.js b/src/web/theme/modules/Search/SearchBar/Autocomplete/Autocomplete.js
+index 9bc108484..ea9a2c91f 100644
+--- a/src/web/theme/modules/Search/SearchBar/Autocomplete/Autocomplete.js
++++ b/src/web/theme/modules/Search/SearchBar/Autocomplete/Autocomplete.js
+@@ -3,10 +3,9 @@ import PropTypes from "prop-types";
+ import { FormattedMessage } from "react-intl";
+ import ProductsResults from "./ProductsResults";
+ import CategoriesResults from "./CategoriesResults";
+-import PagesResults from "./PagesResults";
```

Comment:
suggestion not related to the doc

Wouldn't it be better to manage the full autocomplete search in a Loader server-side and return the result to the app ? 

This way an environment variable would be enough to disable some search features like CmsPages

Reply:
you need to update the client query anyways. so why search everything when not everything is needed.

the thing we should do imo is to ship the search bar with cms page disabled by default